docs(consensus::transaction): Add motivation for TX-validity (#163)
Neptune-Crypto/neptune-core: docs/src/consensus/transaction.md

## Validity

Transaction validity is designed to check four conditions:

> 1. All involved lockscripts halt gracefully

Reviewer: ```suggestion
1. The lock scripts of all input UTXOs halt gracefully
```

> Transaction validity is designed to check four conditions:
> 1. All involved lockscripts halt gracefully
> 2. All involved typescripts halt gracefully
> 3. All input UTXOs (except for an optional coinbase) are present in the append-only commitment list
> 4. All input UTXOs are *not* present in the sliding-window Bloom filter.

Reviewer: ```suggestion
Transaction validity is designed to check four conditions:
1. The lock scripts of all input UTXOs halt gracefully
2. All involved typescripts halt gracefully
3. All input UTXOs are present in the mutator set's append-only commitment list
4. All input UTXOs are *not* present in the mutator set's sliding-window Bloom filter.
```
